workflow/controller/controller.go

```diff
@@ -624,7 +579,26 @@ func getWfPriority(obj interface{}) (int32, time.Time) {
 	return int32(priority), un.GetCreationTimestamp().Time
 }
 
+func getWfPhase(obj interface{}) wfv1.NodePhase {
+	un, ok := obj.(*unstructured.Unstructured)
+	if !ok {
+		return ""
+	}
+	phase, hasPhase, err := unstructured.NestedString(un.Object, "status", "phase")
+	if err != nil {
+		return ""
+	}
+	if !hasPhase {
+		return wfv1.NodePending
+	}
+	return wfv1.NodePhase(phase)
+}
+
 func (wfc *WorkflowController) addWorkflowInformerHandler() {
+	metricsEventHandler := wfc.getMetricsEventHandler()
+	wfc.completedWfInformer.AddEventHandler(metricsEventHandler)
+	wfc.incompleteWfInformer.AddEventHandler(metricsEventHandler)
```

Comment:
These two informers are mutually exclusive... adding the handler to both should ensure that metrics events are only processed once.

Reply:
when I investigated - I found that the informers call DeleteFunc not when the object was deleted  - as you'd expect - but just when it was deleted from the informer - so the behaviour was wrong for this use case

I think we should have a single informer and uses a filter func for the incomplete event handler

Reply:
Will do